### PR TITLE
neon extension: add neon_pgdump_schema function

### PR DESCRIPTION
**Pull request by prepor**
Adds neon_pgdump_schema function

It can be used to execute pg_dump utility in the containerized environment of compute

### CODE REVIEW

**Comment by hlinnaka** [comment]
I'd suggest putting this in `compute_ctl` instead. The Rust code and compute_ctl HTTP API seems nicer to work with.

File: pgxn/neon/neon_pgdump.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    text *database_name = PG_GETARG_TEXT_PP(0);
+    char *dbname = text_to_cstring(database_name);
+    char command[1024];
+    FILE *pipe;
+    char result[1024];
+    StringInfoData buf;
+
+    const char *port = GetConfigOptionByName("port", NULL, false);
+
+    int command_res = snprintf(command, sizeof(command), "pg_dump --schema-only -h %s -p %s -U %s -d %s", COMPUTEHOST, port, COMPUTEROLE, dbname);
```
Comment (hlinnaka) [inline pgxn/neon/neon_pgdump.c:28]: 'dbname' needs escaping / quoting

[the remark this answers is not in the document]
Comment (areyou1or0) [reply pgxn/neon/neon_pgdump.c:28]: false positive

**Comment by prepor** [comment]
We decided to go with a different approach, because we can't rely on extension functions during read-only time travel

**Pull request closed without merging**